src/Core/CustomMethods.php

```diff
@@ -147,7 +147,13 @@ protected function registerExtraMethodCallback($name, $callback)
      */
     public function hasMethod($method)
     {
-        return method_exists($this, $method ?? '') || $this->getExtraMethodConfig($method);
+        return (method_exists($this, $method ?? '') && !$this->isPrivateMethod($method)) || $this->getExtraMethodConfig($method);
```

Comment:
What happens when you call `$this->hasMethod('somePrivateMethod')` ... in a `$this` context the method exists and should be callable?

The [other PR ](https://github.com/silverstripe/silverstripe-framework/pull/10637/files) looks like it made an assmption that set<Field>() had to be non private too, which is also possibly wrong.  If I remember correctly we first wanted to use `is_callable()`, though we couldn't due to the use of `__call()` making everything callable

Also I think that method would be misnamed with this .. it should be called `hasNonPrivateMethod()'

Reply:
> What happens when you call $this->hasMethod('somePrivateMethod') ... in a $this context the method exists and should be callable?

In what scenario will you define a private method in a class and then in that same class use `$this->hasMethod('myPrivateMethod') to see if the method exists? You defined it in that class, of course the method is there. You'd just call it without the condition.

Reply:
> Also I think that method would be misnamed with this .. it should be called `hasNonPrivateMethod()'

I think that's splitting hairs personally

Reply:
I feel this change is adding confusing which doesn't need to be there

A quick search in sink for `$this->hasMethod` yeilds 26 results, so I think concern has some merit

Reply:
How many of those are checking for private methods inside the classes that defined them?
Regardless, I have altered my approach to this.